forward/fwd.go

```diff
@@ -177,7 +187,18 @@ func (f *httpForwarder) serveHTTP(w http.ResponseWriter, req *http.Request, ctx
 			stream = contentType == "text/event-stream"
 		}
 	}
-	written, err := io.Copy(newResponseFlusher(w, stream), response.Body)
+	_, err = io.Copy(newResponseFlusher(w, stream), response.Body)
+	if err != nil {
+		ctx.log.Errorf("Error copying upstream response Body: %v", err)
+		ctx.errHandler.ServeHTTP(w, req, err)
+		return
+	}
+
+	//No defer on body close in order to force announce trailer after body close
+	response.Body.Close()
```

Comment:
In fact, we don't need the response to be closed, we just need that the body was read until EOF

https://github.com/golang/go/blob/master/src/net/http/response.go#L96